src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -192,8 +192,133 @@ static status_t send_del_route_to_routes_conn(private_kernel_super_ipsec_t* this
+	oper_data.dh_group = curr_ike_cfg->get_dh_group(curr_ike_cfg);
+	oper_data.anti_replay_window_size = child_sa->get_config(child_sa)->get_replay_window(child_sa->get_config(child_sa));
+	oper_data.version = ike_sa_id->get_ike_version(ike_sa_id);
+
+	oper_data.inbound_ipsec_spi = child_sa->get_spi(child_sa, TRUE);
+	oper_data.outbound_ipsec_spi = child_sa->get_spi(child_sa, FALSE);
+	oper_data.inbound_ike_spi = ike_sa_id->get_my_spi(ike_sa_id);
+	oper_data.outbound_ike_spi = ike_sa_id->get_other_spi(ike_sa_id);
+	oper_data.nat_identified = child_sa->has_encap(child_sa);
+	oper_data.status = ike_sa->get_state(ike_sa);
+	oper_data.activeness_status = ike_sa->get_is_active(ike_sa);
+
+	char device_public_addr[MAX_IP_STR_LEN];
+	char remote_public_addr[MAX_IP_STR_LEN];
+	char local_network[MAX_IP_PREFIX_STR_LEN];
+	if(! data->src->address_to_string(data->src, device_public_addr)) {
+		LOG(DBG_KNL, SL_LOG_ERR, "failed to get device public address");
+		goto out;
+	}
+	if(! data->src->address_to_string(data->dst, remote_public_addr)) {
+		LOG(DBG_KNL, SL_LOG_ERR, "failed to get remote public address");
+		goto out;
+	}
+
+	struct pm_proto_prefix prefix;
+	if(! ts_to_picomsg_prefix(id->dst_ts, &prefix)) {
+		LOG(DBG_KNL, SL_LOG_ERR, "failed to get local network");
+		goto out;
+	}
+	if(! pm_proto_addr_to_string(&prefix.prefix, local_network)) {
+		LOG(DBG_KNL, SL_LOG_ERR, "failed to get convert local address to string");
+		goto out;
+	}
+
+	oper_data.device_public_address = device_public_addr;
+	oper_data.remote_public_address = remote_public_addr;
+	oper_data.device_public_src_port = data->src->get_port(data->src);
+	oper_data.local_network = local_network;
+
+	oper_data.account_id = super_peer_info->get_account_id(super_peer_info);
+	oper_data.vrf_name = super_peer_info->get_vrf_name(super_peer_info);
+	oper_data.device_id = super_peer_info->get_device_id(super_peer_info);
+
+	struct pm_terminator_conn* terminator_conn = NULL;
+	terminator_conn = lookup_terminator_conn(this, _get_sp_lookup_addr(data, id));
+	if(! terminator_conn) {
+		LOG(DBG_KNL, SL_LOG_ERR, "terminator not found");
+		goto out;
+	}
+	oper_data.term_id = picomsg_terminator_conn_get_id(terminator_conn);
+	picomsg_terminator_conn_unref(terminator_conn);
+
+	const struct system_enterprise_map_lookup_params params = {
+		.terminator_id = oper_data.term_id,
+		.tenant_id = super_peer_info->get_tenant_id(super_peer_info),
+		.account_id = oper_data.account_id,
+		.vrf_name = oper_data.vrf_name,
+	};
+
+	struct system_enterprise_map_enterprise enterprise;
+	if (! system_enterprise_map_lookup_enterprise(this->system_enterprise_map, &params, &enterprise)) {
+		LOG(DBG_KNL, SL_LOG_ERR, "peer information (terminator_id=%u, tenant_id=%s, account_id=%s, vrf_name=%s) doesn't exist in system info map",
+			oper_data.term_id, super_peer_info->get_tenant_id(super_peer_info), oper_data.account_id, oper_data.vrf_name);
+			goto out;
+	}
+	oper_data.neigh_sub_interface = enterprise.neigh_sub_interface;
```

Comment:
If you need the `neigh_sub_interface` then you need to store it in `ike_super_peer_info`. Otherwise in case where the enterprise is deleted with an active SA and then that SA is e.g. getting rekeyed, this logic will fail to write to oper since `system_enterprise_map_lookup_enterprise()` will fail.